src/installer.ts

```diff
@@ -188,20 +188,33 @@ async function unzipJavaDownload(
 function getDownloadInfo(
   refs: string[],
   version: string,
+  arch: string,
   javaPackage: string
 ): {version: string; url: string} {
   version = normalizeVersion(version);
+
+  let archExtension = '';
+  if (arch === 'x86') {
+    archExtension = 'i686';
+  } else if (arch === 'x64') {
+    archExtension = 'x64';
+  } else {
+    throw new Error(`architecture "${arch}" is not int [x86 | x64]`);
```

Comment:
We should validate that the architecture input is valid when we retrieve the input in https://github.com/actions/setup-java/blob/master/src/setup-java.ts#L12

Then this whole block can just be:
```
const archExtension = arch === 'x86' ? 'i686' : 'x64';
```

Reply:
Indeed. I'm not very familiar with the GitHub Actions core library (and JS in general) so bear with me. I'll make the change soon.

Reply:
No worries! Let me know if you need any help

Reply:
Should be "not in" instead of "not int".